src/hotspot/cpu/ppc/c1_LIRGenerator_ppc.cpp

```diff
@@ -885,7 +893,14 @@ void LIRGenerator::do_NewInstance(NewInstance* x) {
     tty->print_cr("   ###class not loaded at new bci %d", x->printable_bci());
   }
 #endif
-  CodeEmitInfo* info = state_for(x, x->state());
+  CodeEmitInfo* info = nullptr;
+  if (x->state_before() != nullptr && x->state_before()->force_reexecute()) {
+    info = state_for(x, x->state_before());
+    info->set_force_reexecute();
+  } else {
+    info = state_for(x, x->state());
+  }
+
```

Comment:
This code seems to be integrated at the wrong place. Other platforms have it in `LIRGenerator::do_NewTypeArray`.